packages/cli/src/TestWebhooks.ts

```diff
@@ -253,17 +254,25 @@ export class TestWebhooks implements IWebhookManager {
 
 			cacheableWebhook.userId = userId;
 
+			const registration: TestWebhookRegistration = {
+				sessionId,
+				workflowEntity,
+				destinationNode,
+				webhook: cacheableWebhook as IWebhookData,
+			};
+
 			try {
+				/**
+				 * Register the test webhook _before_ creation at third-party service
+				 * in case service sends a confirmation request immediately on creation.
+				 */
+				await this.registrations.register(registration);
+
 				await workflow.createWebhookIfNotExists(webhook, NodeExecuteFunctions, 'manual', 'manual');
 
 				cacheableWebhook.staticData = workflow.staticData;
 
-				await this.registrations.register({
-					sessionId,
-					workflowEntity,
-					destinationNode,
-					webhook: cacheableWebhook as IWebhookData,
-				});
+				await this.registrations.register(registration);
```

Comment:
We can now remove this row right? And if it fails, shouldn't we deregister as part of the catch block?

Reply:
This line ensures the static data that was updated during creation makes it into the cache.

Reply:
Got it! What about the automated de-registration if it fails?

Reply:
If registration throws, catch calls `deactivateWebhooks` where the last line deregisters test webhooks. That was not changed in this PR.